### PR TITLE
Improve resource usage when NFS is in high load.

### PR DESCRIPTION
**Pull request by araujobsd**
- wrapper_read() to ensure the file stays open as minimum as possible

### CODE REVIEW

**Comment by araujobsd** [comment]
@diconico07 I think now it is safer and with the wrapper_read() it spends less time with the file open.

If after these changes we still have a decrease in performance, an alternative is to copy the files out of procfs to a temporary place and parse it from there.

**Review by diconico07 — CHANGES_REQUESTED** [review]
This looks good to me overall, just the cgroups part that looks dangerous to me.
If you confirm it is not messing with existing cgroup hierarchy usage then I'll approve this.

File: src/main.rs [hunk 1]
```diff
@@ -47,11 +50,31 @@ fn help() {
...
 }
 
+fn set_cpu_percent(percent: u64) {
+    let cg_nfsv4_exporter = cgroups_rs::hierarchies::auto();
+    let cg: Cgroup = CgroupBuilder::new("prometheus-nfsv4-exporter")
+        .cpu().shares(percent).done().build(cg_nfsv4_exporter);
+
+    let cg_cpus: &cgroups_rs::cpu::CpuController = cg.controller_of().unwrap();
+    let my_pid = std::process::id() as u64;
+    let cgroup_setup = cg_cpus.add_task(&CgroupPid::from(my_pid));
```
[the remark this answers is not in the document]
Comment (araujobsd) [reply src/main.rs:60]: It should not! But, I agree we should not deal with CGROUP here, it is not the purpose of this software.
I will remove this CGROUP code.

[the remark this answers is not in the document]
Comment (araujobsd) [reply src/main.rs]: I have removed the CGROUP, as I don't think the export should deal with that by itself. Thanks to point it out @diconico07

**Pull request merged**